src/voice_public.js

```diff
@@ -11,19 +11,37 @@ go.app = function() {
         App.call(self, 'state_start');
         var $ = self.$;
         var lang = 'eng_NG';
-        var interrupt = true;
 
         self.add = function(name, creator) {
             self.states.add(name, function(name, opts) {
-                if (!interrupt || !go.utils_project.should_restart(self.im))
-                    return creator(name, opts);
-
-                interrupt = false;
-                opts = opts || {};
-                opts.name = name;
-                // Prevent previous content being passed to next state
-                self.im.msg.content = null;
-                return self.states.create('state_start', opts);
+                if (go.utils_project.should_repeat(self.im)) {
+                    // ensure that when the user requested a repeat by '*' when
+                    // in a retry state that the original state will rather be repeated
+                    if (opts.retry) opts.retry = false;
```

Comment:
I'd rather we not complicate this behaviour further by adding this.  If they want to replay the message they should hear the retry message.